incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaPublishFactory.java

```diff
@@ -357,16 +359,16 @@ private void onMqttData(
 
             if (mqttPublishDataEx.responseTopic().asString() != null)
             {
-                final String16FW replyTopic = mqttPublishDataEx.responseTopic();
+                final String16FW responseTopic = mqttPublishDataEx.responseTopic();
                 addHeader(helper.kafkaReplyToHeaderName, kafkaMessagesTopic);
-                addHeader(helper.kafkaReplyKeyHeaderName, replyTopic);
+                addHeader(helper.kafkaReplyKeyHeaderName, responseTopic);
 
-                String[] replyTopicHeaders = replyTopic.asString().split("/");
-                for (int i = 0; i < replyTopicHeaders.length; i++)
+                AtomicInteger offset = new AtomicInteger();
+                Arrays.stream(responseTopic.asString().split("/")).forEach(segment ->
                 {
-                    final String16FW replyTopicHeader = new String16FW(replyTopicHeaders[i]);
-                    addHeader(helper.kafkaReplyFilterHeaderName, replyTopicHeader);
-                }
+                    final int length = segment.length();
```

Comment:
`String.length()` and bytes length are not the same for any multi-byte UTF-8 characters in the string.
It may be easier to find the `/` (byte) in the underlying bytes than convert to String, split on '/' and then convert back to bytes.